app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java

```diff
@@ -0,0 +1,67 @@
+package com.mobile.tandil.javabaseproject.mvp.presenter;
+
+import com.mobile.tandil.javabaseproject.R;
+import com.mobile.tandil.javabaseproject.entity.Reservation;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+import com.mobile.tandil.javabaseproject.util.DateUtils;
+import com.mobile.tandil.javabaseproject.util.ValidationEnums;
+import java.util.Calendar;
+
+public class ReservationPresenter implements ReservationContract.Presenter {
+
+    private final ReservationContract.View view;
+    private final ReservationContract.Model model;
+
+    public ReservationPresenter(ReservationContract.View view, ReservationContract.Model model) {
+        this.view = view;
+        this.model = model;
+    }
+
+    @Override
+    public void showPickers(ListenerPickers listenerPickers, Boolean startOrEnd) {
+        view.showDateAndTime(listenerPickers, startOrEnd);
+    }
+
+    @Override
+    public void listenStartingDate(Calendar startingDate) {
+        view.showStartingDate(DateUtils.calendarToString(startingDate));
+    }
+
+    @Override
+    public void listenEndingDate(Calendar endingDate) {
+        view.showEndingDate(DateUtils.calendarToString(endingDate));
+    }
+
+    @Override
+    public void confirmReservation(Reservation reservation, String strPark) {
+        Calendar startingDate = reservation.getStartingDate();
+        Calendar endingDate = reservation.getEndingDate();
+        String key = reservation.getKey();
+
+        if (!key.isEmpty() && !strPark.isEmpty() && startingDate != null && endingDate != null) {
+            Integer parkingNumber = Integer.parseInt(strPark);
+            ValidationEnums rta = model.validations(startingDate, endingDate, parkingNumber);
+            switch (rta) {
+                case START_BEFORE_NOW:
+                    view.showToastInReservation(R.string.start_date_before_now);
```

Comment:
The presenter must not know the resources. The view should have a method for each error.

Reply:
done